apparmor.d/profiles-s-z/vesktop

```diff
@@ -0,0 +1,46 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2024 odomingao
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+include <tunables/global>
+
+@{name} = vesktop
+@{lib_dirs} = @{lib}/@{name}
+@{config_dirs} = @{user_config_dirs}/@{name}
+@{cache_dirs} = @{user_cache_dirs}/@{name}
+
+@{exec_path} = @{bin}/vesktop
+profile vesktop @{exec_path} flags=(attach_disconnected) {
+  include <abstractions/base>
+  include <abstractions/audio-client>
+  include <abstractions/common/electron>
+  include <abstractions/p11-kit>
+  include <abstractions/user-download-strict>
+  include <abstractions/video>
+
+  network inet dgram,
+  network inet6 dgram,
+  network inet stream,
+  network inet6 stream,
+  network netlink raw,
+
+  @{exec_path} mr,
+
+  owner /tmp/.org.chromium.Chromium.@{rand6} mr,
+  owner @{run}/user/@{uid}/discord-ipc-@{int} rw,
+
+  @{sys}/devices/@{pci}/usb@{int}/**/interface r,
+
+        @{PROC}/@{pid}/cmdline r,
+  owner @{PROC}/@{pid}/task/@{tid}/comm rw,
+
+  owner /dev/ r,
+
+  deny       /dev/tty rw,
+  deny owner /dev/tty@{int} rw,
```

Comment:
Why do you deny them?

Reply:
As a silencer. It asks for these permissions but I don't think that it actually needs them, as everything on the program works fine, and AFAIK (correct me if I'm wrong) access to /dev/tty allows the program to escape confinement so I thought it was better to deny.

Reply:
Do you have any source of this?

Reply:
For the possible /dev/tty escapes? https://www.errno.fr/TTYPushback.html and https://github.com/containers/bubblewrap/issues/142

Reply:
Thanks. This has been discussed here: https://github.com/roddhjav/apparmor.d/discussions/340

Spoiler alert: the solution is not apparmor.